text/0000-deprecate-some-utils.md

```diff
@@ -0,0 +1,57 @@
+- Start Date: 2018-05-22
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Deprecate Existence Utils & Computed Properties
+
+## Summary
+
+This RFC suggests the deprecation of the following [methods](http://api.emberjs.com/ember/release/modules/@ember%2Futils):
+- `Ember.isEmpty`
+- `Ember.isBlank`
+- `Ember.isNone`
+- `Ember.isPresent`
+
+As well as [computed properties](http://api.emberjs.com/ember/release/modules/@ember%2Fobject) that rely on these methods:
+- `notEmpty`
+- `empty`
+- `none`
+
+Once deprecated, these methods will move to an external addon to allow users to still take advantage of the succinctness of these utility methods.
+
+## Motivation
+
+To further Ember's goal of a svelte core and moving functionality to separate packages, this RFC attempts to deprecate a few utility methods and extract them to their own addon.  A part from the aforementioned goal, in many cases, these utility methods do not add a whole lot of value over writing plain old JavaScript.  Moreover, these methods generally have a negative cognitive impact on the code in question.  In order to reduce code liability and advocate for users to write plain JavaScript, these methods could be removed from Ember core and extracted to an addon.
+
+A simple example of this cognitive load would be checking for the presence of a variable of type String that can only have four states - `undefined`, `null`, `''` or `'MYSTRING'`.  One might check if the variable is truthy with `!!myVariable`.  However, a utility method such as `isPresent(myVariable)` can increase the cognitive load of the truthy check in question as it checks for many different types.
+
+In addition, when checking whether a primitive value is truthy, a user can choose from `isPresent`, `isBlank`, `isNone`, or `isEmpty`.  This can can lead to some confusion for users as there are many options to do the same thing.
+
+Lastly, `Ember.isEmpty` documented [here](https://www.emberjs.com/api/ember/release/functions/@ember%2Futils/isEmpty) will return false for an empty object.  Defining what an empty object is can be a bit tricky and `isEmpty` provided specific semantics that may not fit a users definition.  This idea of specific semantics that do not fit a users definition also applies to the other utility methods as well.
+
+## Detailed design
+
+`Ember.isEmpty`, `Ember.isBlank`, `Ember.isNone`, `Ember.isPresent`, `notEmpty`, `empty`, and `none` will be deprecated at first with an addon that extracts these methods for users that still find value in these utility methods.  Any instantiation of one of these functions will log a deprecation warning, notifying the user that this method is deprecated and provide a link to the supported addon.
+
+Addons that previously relied on these utility methods would also be expected to install this package. Moreover, to ease adoption of the external addon, a codemod will become available for the community.
```

Comment:
I'm also in favor of answering @simonihmig's question here about what the codemod would do (replacing the code or updating the import). Either is acceptable, IMO.